languages/python/tests/test_atom.py

```diff
@@ -2118,6 +2118,79 @@ def test_reference_ignore_serialization(self, caller):
         assert success == True
         assert len(failed) == 0
 
+    # The input data comes in 3 forms:
+    # large, multiple large, and multiple small
+    @pytest.mark.parametrize(
+        "data",
+        [
+            [
+                np.array(
+                    [[[0, 255, 255] for n in range(0, 2000)] for i in range(0, 4000)],
+                    dtype=np.uint8,
+                )
+            ],
+            [
+                np.array(
+                    [[[r, r, r] for n in range(0, 2000)] for i in range(0, 4000)],
+                    dtype=np.uint8,
+                )
+                for r in range(0, 5)
+            ],
+            [
+                np.array(
+                    [[[r, r, r] for n in range(0, 5)] for i in range(0, 20)],
+                    dtype=np.uint8,
+                )
+                for r in range(0, 100)
+            ],
+        ],
+    )
+    def test_reference_compare_timing_large(self, caller, data):
+        caller, caller_name = caller
+
+        before_shm_create = time.time_ns()
+        success, ref_ids = caller.reference_create(
+            *data, ref_type=AtomRefType.SHARED_MEM
+        )
+        assert success
+        after_shm_create = time.time_ns()
+        get_data = caller.reference_get(*ref_ids)
+        assert np.array_equal(get_data, data)
+        after_shm_get = time.time_ns()
+        success, failed = caller.reference_delete(*ref_ids)
+        assert success and len(failed) == 0
+        after_shm_delete = time.time_ns()
+
+        before_redis_create = time.time_ns()
+        success, ref_ids = caller.reference_create(
+            *data, ref_type=AtomRefType.REDIS, serialization="arrow"
+        )
+        assert success
+        after_redis_create = time.time_ns()
+        get_data = caller.reference_get(*ref_ids)
+        assert np.array_equal(get_data, data)
+        after_redis_get = time.time_ns()
+        success, failed = caller.reference_delete(*ref_ids)
+        assert success and len(failed) == 0
+        after_redis_delete = time.time_ns()
+
+        shm_create = after_shm_create - before_shm_create
+        redis_create = after_redis_create - before_redis_create
+        shm_get = after_shm_get - after_shm_create
+        redis_get = after_redis_get - after_redis_create
+        shm_delete = after_shm_delete - after_shm_get
+        redis_delete = after_redis_delete - after_redis_get
+
+        # shm is within 10% of the time it takes for redis
+        # to complete reference create, get, and delete
+        assert (
+            shm_create <= 1.1 * redis_create
```

Comment:
Maybe this check should be something like this instead:
```suggestion
            shm_create - redis_create <= 1.1 * min(shm_create, redis_create)
```
so we are checking that the difference between the times is within 10% of which ever is smaller? Since as written, if the redis time became very large by mistake, the current check would still pass and we'd miss the regression.

Reply:
Not sure if 10% is the right number, maybe its not and maybe we should also be checking a hard threshold in case they _both_ increased drastically.

Reply:
I think we should just have a hard threshold here, maybe 100ms, around a fairly large piece of data. Making it relative will be asking for trouble/flaky tests, and we really care more about not adding some undue overhead.

Reply:
Thanks for the feedback on that Dan. Do we care about the timing when running tests locally vs circle because the times are looking pretty dramatically different